### PR TITLE
Added/Modified some features

### PR DESCRIPTION
**Pull request by amandaJayanetti**

### CODE REVIEW

File: cleanup [hunk 1]
```diff
@@ -0,0 +1,7 @@
+#!/bin/sh
+
+rm -rf	autom4te.cache/ config.* src/Makevars \
+    	src/*.o src/*.d src/*.a src/*.dll src/*.so \
+    	*/*~ *~ 
+
+
```
Comment (jaylett) [inline cleanup]: You should avoid the trailing lines here. Also, scripts need to be `chmod +x` before you commit them (git tracks permissions as well). You can add both in as a fixup.

**Comment by jaylett** [comment]
@amandaJayanetti when I said do a fixup commit, I wasn't as clean as I could have been — sorry. What you can do is this:

```
git commit --fixup <sha>
```

with `<sha>` that of a previous commit and it'll create a commit with a message that says 'fixup! <previous commit>'. Then you can later do `git rebase -i --autosquash` and it's easy to collapse the two commits into one.

The general rule of thumb is that each commit should be a single, complete, change. So fixup commits are a good way of making corrections to earlier commits during PR (so others can quickly see what a small commit that isn't an entire change is there for), then you can rebase before merging to end up with the history containing only suitable commits.

You can do this manually by using `git rebase -i` and then moving the line for a fixup commit so it's immediately after the line for the previous commit, and changing 'pick' to 'fixup' at the start of the fixup commit's line. (Interactive rebase, which is what the `-i` flag does, is very powerful, but this is probably the simplest use of it — re-ordering commits and collapsing them together.)

**Comment by amandaJayanetti** [comment]
Done :)

**Comment by jaylett** [comment]
Let's merge this then :)

Just as a warning: I'll keep on badgering you about making each commit do one complete thing, and push you to write the best possible commit messages explaining each. (There's some suggestions in [the Xapian developer guide](https://xapian-developer-guide.readthedocs.io/en/latest/contributing/workflow.html#make-commits-out-of-your-changes) on good commits and good commit messages, but feel free to ask if anything's unclear!)

**Pull request merged**